### PR TITLE
Improved error message in the `from_dict`

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/19410

Background: 
@jongio's code tried sending EG event schema to cloud event's from_dict method which won't work. Need to imrpove the error message here.

### CODE REVIEW

File: sdk/core/azure-core/azure/core/messaging.py [hunk 1]
```diff
@@ -122,6 +122,16 @@ def from_dict(cls, event):
         :type event: dict
         :rtype: CloudEvent
         """
+        if not all([_ in event for _ in ("source", "type")]):
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/messaging.py:125]: Can you add the description of the doc or a link?

File: sdk/core/azure-core/azure/core/messaging.py [hunk 2]
```diff
@@ -122,6 +122,16 @@ def from_dict(cls, event):
...
+                raise ValueError(
+                    "The event does not conform to the cloud event spec." + 
+                    " Try using the EventGridEvent from azure-eventgrid library"
+                )
+            raise ValueError(
+                "The event does not conform to the cloud event spec. source and type are required."
+                )
+
         kwargs = {}  # type: Dict[Any, Any]
         reserved_attr = [
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/messaging.py:136]: Are these keys case-sensitive?

Comment (rakshith91) [reply sdk/core/azure-core/azure/core/messaging.py:138]: yes

File: sdk/core/azure-core/azure/core/messaging.py (hunk 1 above)
Comment (rakshith91) [reply sdk/core/azure-core/azure/core/messaging.py:127]: added https://github.com/cloudevents/spec

File: sdk/core/azure-core/azure/core/messaging.py [hunk 3]
```diff
@@ -122,6 +122,19 @@ def from_dict(cls, event):
         :type event: dict
         :rtype: CloudEvent
         """
+        # https://github.com/cloudevents/spec Cloud event spec requires source, type,
+        # specversion. We autopopulate everything other than source, type.
+        if not all([_ in event for _ in ("source", "type")]):
+            if all([_ in event for _ in ("subject", "eventType", "data", "dataVersion", "id", "eventTime")]):
+                raise ValueError(
+                    "The event does not conform to the cloud event spec https://github.com/cloudevents/spec." +
+                    " Try using the EventGridEvent from azure-eventgrid library"
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/messaging.py:131]: I would be more explicit:
It looks like your event is an EventGrid schema. You can parse EventGrid events using EventGridEvent.from_dict of the azure-eventgrid library

File: sdk/core/azure-core/azure/core/messaging.py [hunk 4]
```diff
@@ -122,6 +122,19 @@ def from_dict(cls, event):
...
+                )
+            raise ValueError(
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/messaging.py:133]: This is client side validation. I would do this whole block into a try/except around the `cls` at the end

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**